stubs/iterable.stub

```diff
@@ -298,3 +296,28 @@ class AppendIterator extends IteratorIterator {
     public function getArrayIterator() {}
 
 }
+
+/**
+ * @template-covariant TKey
+ * @template-covariant TValue
+ * @template TIterator as Iterator<TKey, TValue>
+ *
+ * @template-extends IteratorIterator<TKey, TValue, TIterator>
+ */
+class NoRewindIterator extends IteratorIterator {
```

Comment:
This stub is different from the one in Psalm: https://github.com/vimeo/psalm/blob/36d5a2a83c1d7b3e2aca1cfe9e9dcc916619e962/stubs/CoreGenericIterators.phpstub#L581-L604 (This one uses three template types, Psalm uses two). We need to be cross-compatible here.

Reply:
Note that NoRewindIterator has three templates in the next major of Psalm: https://github.com/vimeo/psalm/blob/master/stubs/CoreGenericIterators.phpstub#L593

(as well as every child of IteratorIterator IIRC)

Reply:
Indeed, I inspired those PRs from what's in PSalm.